### PR TITLE
search impl

### PR DESCRIPTION
**Pull request by tikitko**

### CODE REVIEW

File: blog-server-services/src/impls/rbatis_post_service.rs [hunk 1]
```diff
@@ -82,6 +92,32 @@ impl Post {
...
+            post.*, \
+            author.slug AS author_slug, \
+            author.first_name AS author_first_name, \
+            author.last_name AS author_last_name, \
+            string_agg(concat_ws(',', tag.slug, tag.title), ';') as tags \
+        FROM post \
+        JOIN author ON post.author_id = author.id \
+        LEFT JOIN post_tag ON post_tag.post_id  = post.id \
+        LEFT JOIN tag ON tag.id = post_tag.tag_id \
+        WHERE post.title LIKE '%#{query}%' OR post.summary LIKE '%#{query}%' OR post.content LIKE '%#{query}%' \
```
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:107]: '%#{query}%' is not working
🤡🤡🤡🤡🤡
CC: @YoshkiMatryoshki

**Comment by tikitko** [comment]
@YoshkiMatryoshki ready for review

File: blog-server-services/src/impls/rbatis_author_service.rs [hunk 2]
```diff
@@ -7,9 +7,14 @@ pub fn create_rbatis_author_service(rb: RBatis) -> Box<dyn AuthorService> {
...
-impl_select!(Author {select_by_slug(slug: &String) -> Option => "`WHERE slug = #{slug} LIMIT 1`"});
-impl_select!(Author {select_all_with_offset_and_limit(offset: &i64, limit: &i64) => "`LIMIT #{limit} OFFSET #{offset}`"});
+impl_select!(Author {select_by_id(id: &i64) -> Option => 
+    "`WHERE id = #{id} LIMIT 1`"});
+impl_select!(Author {select_by_slug(slug: &String) -> Option => 
+    "`WHERE slug = #{slug} LIMIT 1`"});
+impl_select!(Author {select_all_with_offset_and_limit(offset: &i64, limit: &i64) => 
+    "`LIMIT #{limit} OFFSET #{offset}`"});
+impl_select!(Author {select_all_by_query_with_offset_and_limit(query: &String, offset: &i64, limit: &i64) => 
+    "`WHERE author.slug ILIKE '%' || #{query} || '%' OR author.first_name ILIKE '%' || #{query} || '%' OR author.middle_name ILIKE '%' || #{query} || '%' OR author.last_name ILIKE '%' || #{query} || '%' LIMIT #{limit} OFFSET #{offset}`"});
```
Comment (YoshkiMatryoshki) [inline blog-server-services/src/impls/rbatis_author_service.rs:17]: There is a possibility that 2 users will change let say their middlenames after our first pagination call and before second, so each time we will obtain different set of data. 

i dont think we have workaroung for now and for MVP could be ok, but maybe we should take it in consideration in the future.

Also im not sure that order of records will be consistent

**Review by YoshkiMatryoshki — APPROVED** [review]

File: blog-server-services/src/impls/rbatis_author_service.rs (hunk 2 above)
Comment (tikitko) [reply blog-server-services/src/impls/rbatis_author_service.rs:17]: - "There is a possibility that 2 users will change let say their middlenames after our first pagination call and before second, so each time we will obtain different set of data."
**It's expected I believe, same situation for all other places with pagination (posts moved on other pages and etc.)**
- i dont think we have workaroung for now and for MVP could be ok, but maybe we should take it in consideration in the future.
**Will create task.**
- Also im not sure that order of records will be consistent
Good point, will check.

Comment (tikitko) [reply blog-server-services/src/impls/rbatis_author_service.rs:17]: - [ ] Check "Also im not sure that order of records will be consistent"

Comment (tikitko) [reply blog-server-services/src/impls/rbatis_author_service.rs:17]: Looks sorted as placed in database, for now applicable.

**Pull request merged**